posthog/api/plugin.py

```diff
@@ -203,15 +208,55 @@ def check_for_updates(self, request: request.Request, **kwargs):
 
         return Response({"plugin": PluginSerializer(plugin).data})
 
+    @action(methods=["GET"], detail=True)
+    def source(self, request: request.Request, **kwargs):
+        plugin = self.get_plugin_with_permissions(reason="source editing")
+        response: Dict[str, str] = {}
+        for source in PluginSourceFile.objects.filter(plugin=plugin):
+            response[source.filename] = source.source
+        return Response(response)
+
+    @action(methods=["PATCH"], detail=True)
+    def update_source(self, request: request.Request, **kwargs):
+        plugin = self.get_plugin_with_permissions(reason="source editing")
+        sources: Dict[str, PluginSourceFile] = {}
+        performed_changes = False
+        for source in PluginSourceFile.objects.filter(plugin=plugin):
+            sources[source.filename] = source
+        for key, value in request.data.items():
+            if not sources.get(key):
+                sources[key] = PluginSourceFile.objects.create(plugin=plugin, filename=key, source=value)
+                continue
+            if sources[key].source != value:
+                performed_changes = True
+                if value is None:
+                    sources[key].delete()
+                    del sources[key]
+                else:
+                    sources[key].source = value
+                    sources[key].save()
+        response: Dict[str, str] = {}
+        for key, source in sources.items():
+            response[source.filename] = source.source
+
+        # Update values from plugin.json
+        plugin_json = json.loads(response.get("plugin.json") or "{}")
+        if plugin_json["name"] != plugin.name:
+            plugin.name = plugin_json["name"]
+            performed_changes = True
+        if json.dumps(plugin_json.get("config") or []) != json.dumps(plugin.config_schema or []):
```

Comment:
Same - use dict.get second argument here.

Reply:
Same reasoning here. `plugin_json.get("config")` will return None, but I want that to be comparable to `[]` in this line, if that's what `plugin.config_schema` contains